virttest/libvirt_xml/vm_xml.py

```diff
@@ -677,14 +677,20 @@ def undefine(self, options=None, virsh_instance=base.virsh):
 
         return virsh_instance.remove_domain(self.vm_name, options)
 
-    def define(self, virsh_instance=base.virsh):
-        """Define VM with virsh from this instance"""
+    def define(self, virsh_instance=base.virsh, detail=False):
```

Comment:
generally, we don't encourage function return two different primitive data type. Here sometimes it will return one bool variable, sometimes return result as one object

Reply:
Updated comments with more detail. Could you help review again? Thanks

Reply:
I understand too many Xml content output happened in the sync(). The natural way is to provide one option in sync (xx) method to suppress or  allow it if define return False. And what's more, suppress can be set to default.

Reply:
That is an option. But whether we add an option to sync() or not, the current change is necessary, or there would not be detailed messaged to return.
And if in the future, anyone would like to have full content of xml printed out as error message, they can modify the function sync() then. Currently I'd like to have a shorter and clearer error message.